From edba536cf9434d3ba362f35686c37a5d751f6f68 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Fri, 3 Jul 2020 16:04:45 +0800
Subject: [PATCH] policy/modules/admin/logwatch: fix avc denials for logwatch

* Add getsched to logwatch_t process
* Allow logwatch_t to create logwatch_lock_t dirs
* Allow logwatch_mail_t to read/write pipe of crond

Fixes:
avc:  denied  { getsched } for  pid=1012 comm="sort"
scontext=system_u:system_r:logwatch_t:s0-s15:c0.c1023
tcontext=system_u:system_r:logwatch_t:s0-s15:c0.c1023 tclass=process
permissive=0

avc:  denied  { write } for  pid=269 comm="lockfile-create"
name="logcheck" dev="tmpfs" ino=12709
scontext=system_u:system_r:logwatch_t:s0-s15:c0.c1023
tcontext=system_u:object_r:logwatch_lock_t:s0 tclass=dir permissive=0

avc:  denied  { write } for  pid=1470 comm="sendmail"
path="pipe:[15133]" dev="pipefs" ino=15133
scontext=system_u:system_r:logwatch_mail_t:s0-s15:c0.c1023
tcontext=system_u:system_r:crond_t:s0-s15:c0.c1023 tclass=fifo_file
permissive=0

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>

Remove sections which have been applied in 2.20221101.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 policy/modules/admin/logwatch.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/admin/logwatch.te b/policy/modules/admin/logwatch.te
index ae02e39b0..a6e884839 100644
--- a/policy/modules/admin/logwatch.te
+++ b/policy/modules/admin/logwatch.te
@@ -186,6 +186,8 @@ dev_read_rand(logwatch_mail_t)
 dev_read_urand(logwatch_mail_t)
 dev_read_sysfs(logwatch_mail_t)
 
+cron_rw_pipes(logwatch_mail_t)
+
 logging_read_all_logs(logwatch_mail_t)
 
 optional_policy(`
-- 
2.17.1

